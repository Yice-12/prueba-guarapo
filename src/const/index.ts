export const SPECIE_LIST = [
  "Human",
  "Alien",
  "Humanoid",
  "Poopybutthole",
  "Mythological",
  "Unknown",
  "Animal",
  "Disease",
  "Robot",
  "Cronenberg",
];

export const GENDER_LIST = ["female", "male", "genderless", "unknown"];
export const STATUS_LIST = ["Alive", "Dead", "Unknown"];
