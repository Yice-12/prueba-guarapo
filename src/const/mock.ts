import { configureStore } from "@reduxjs/toolkit";
import { charactersSlice } from "../store/slices/charactersSlice/charactersSlice";

const mockState = {
  characters: {
    characters: [
      {
        id: 1,
        name: "Rick Sanchez",
        status: "Alive",
        species: "Human",
        type: "",
        gender: "Male",
        origin: {
          name: "Earth (C-137)",
          url: "https://rickandmortyapi.com/api/location/1",
        },
        location: {
          name: "Citadel of Ricks",
          url: "https://rickandmortyapi.com/api/location/3",
        },
        image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
        episode: ["https://rickandmortyapi.com/api/episode/1"],
        url: "https://rickandmortyapi.com/api/character/1",
        created: "2017-11-04T18:48:46.250Z",
      },
    ],
    loading: false,
    character: {
      id: 1,
      name: "Rick Sanchez",
      status: "Alive",
      species: "Human",
      type: "",
      gender: "Male",
      origin: {
        name: "Earth (C-137)",
        url: "https://rickandmortyapi.com/api/location/1",
      },
      location: {
        name: "Citadel of Ricks",
        url: "https://rickandmortyapi.com/api/location/3",
      },
      image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
      episode: ["https://rickandmortyapi.com/api/episode/1"],
      url: "https://rickandmortyapi.com/api/character/1",
      created: "2017-11-04T18:48:46.250Z",
    },
    episodes: [
      {
        id: 1,
        name: "Pilot",
        air_date: "December 2, 2013",
        episode: "S01E01",
        characters: ["https://rickandmortyapi.com/api/character/1"],
        url: "https://rickandmortyapi.com/api/episode/1",
        created: "2017-11-10T12:56:33.798Z",
      },
    ],
    name: "",
  },
};

export const store = configureStore({
  reducer: {
    characters: charactersSlice.reducer,
  },
  preloadedState: mockState,
});
