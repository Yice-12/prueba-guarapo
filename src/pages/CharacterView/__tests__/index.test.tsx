import { fireEvent, render } from "@testing-library/react";
import { CharacterView } from "../";
import { Provider } from "react-redux";
import { store } from "../../../const/mock";
import { BrowserRouter } from "react-router-dom";

test("render CharacterView and back", () => {
  const { asFragment, getByText } = render(
    <Provider store={store}>
      <BrowserRouter>
        <CharacterView />
      </BrowserRouter>
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();

  const buttonBack = getByText("GO BACK");
  fireEvent.click(buttonBack);
});
