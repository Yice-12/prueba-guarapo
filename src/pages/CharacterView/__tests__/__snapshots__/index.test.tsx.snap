// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render CharacterView 1`] = `
<DocumentFragment>
  <div
    class="characterView"
  >
    <div
      class="characterView__content"
    >
      <div
        class="characterView__arrowBack"
      >
        <img
          alt="arrowBack"
          src="[object Object]"
        />
        <p>
          GO BACK
        </p>
      </div>
      <div
        class="characterView__profile"
      >
        <img
          src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
        />
        <h1>
          Rick Sanchez
        </h1>
      </div>
      <div
        class="characterView__container"
      >
        <div
          class="characterView__information"
        >
          <h2>
            Informations
          </h2>
          <ul
            class="characterView__information-list"
          >
            <li>
              <p>
                Gender
              </p>
              <span>
                Male
              </span>
            </li>
            <li>
              <p>
                Status
              </p>
              <span>
                Alive
              </span>
            </li>
            <li>
              <p>
                Specie
              </p>
              <span>
                Human
              </span>
            </li>
            <li>
              <p>
                Origin
              </p>
              <span>
                Earth (C-137)
              </span>
            </li>
            <li>
              <p>
                Type
              </p>
              <span />
            </li>
            <li>
              <p>
                Location
              </p>
              <span>
                Citadel of Ricks
              </span>
            </li>
          </ul>
        </div>
        <div
          class="characterView__episodes"
        >
          <h2>
            Episodes
          </h2>
          <ul
            class="characterView__episodes-list"
          >
            <li>
              <p>
                S01E01
              </p>
              <span>
                Pilot
              </span>
              <span>
                December 2, 2013
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`render CharacterView and back 1`] = `
<DocumentFragment>
  <div
    class="characterView"
  >
    <div
      class="characterView__content"
    >
      <div
        class="characterView__arrowBack"
      >
        <img
          alt="arrowBack"
          src="[object Object]"
        />
        <p>
          GO BACK
        </p>
      </div>
      <div
        class="characterView__profile"
      >
        <img
          src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
        />
        <h1>
          Rick Sanchez
        </h1>
      </div>
      <div
        class="characterView__container"
      >
        <div
          class="characterView__information"
        >
          <h2>
            Informations
          </h2>
          <ul
            class="characterView__information-list"
          >
            <li>
              <p>
                Gender
              </p>
              <span>
                Male
              </span>
            </li>
            <li>
              <p>
                Status
              </p>
              <span>
                Alive
              </span>
            </li>
            <li>
              <p>
                Specie
              </p>
              <span>
                Human
              </span>
            </li>
            <li>
              <p>
                Origin
              </p>
              <span>
                Earth (C-137)
              </span>
            </li>
            <li>
              <p>
                Type
              </p>
              <span />
            </li>
            <li>
              <p>
                Location
              </p>
              <span>
                Citadel of Ricks
              </span>
            </li>
          </ul>
        </div>
        <div
          class="characterView__episodes"
        >
          <h2>
            Episodes
          </h2>
          <ul
            class="characterView__episodes-list"
          >
            <li>
              <p>
                S01E01
              </p>
              <span>
                Pilot
              </span>
              <span>
                December 2, 2013
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
